src/App.tsx

```diff
@@ -1,116 +1,81 @@
-    this.enableButton = this.enableButton.bind(this);
-    this.throwError = this.throwError.bind(this);
-  }
-
-  async componentDidMount() {
-    this.loadShows();
-    this.enableButton();
-  }
-
-  async loadShows() {
-    const { currentPage, isLoading } = this.state;
-
+import { Route, Routes, useNavigate } from 'react-router-dom';
+import Layout from './components/Layout/Layout';
+import MainPage from './components/pages/MainPage/MainPage';
+import NotFound from './components/pages/NotFound/NotFound';
+import DetailsPage from './components/pages/DetailsPage/DetailsPage';
+
+const App: React.FC = () => {
+  const [shows, setShows] = React.useState<ShowData[]>([]);
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<Error | null | unknown>(null);
+  const [itemsPerPage] = React.useState(20);
+  const [searchQuery, setSearchQuery] = React.useState<string>('');
+  const [isShowMoreButtonDisable, setIsShowMoreButtonDisable] = React.useState<boolean>(false);
+  const [currentPage, setCurrentPage] = React.useState(0);
+  const [nextPage, setNextPage] = React.useState<number | null>(null);
+  const [prevPage, setPrevPage] = React.useState<number | null>(null);
+  const [isCardItemsDarked, setIsCardItemsDarked] = React.useState<boolean>(false);
+
+  const navigate = useNavigate();
+
+  const loadShows = async (page: number) => {
     if (isLoading) {
       return;
     }
 
-    this.setState({ isLoading: true });
+    setIsLoading(true);
 
     try {
-      const response = await fetch(`${URL.SHOWS}?page=${currentPage}`);
+      const response = await fetch(`${URL.SHOWS}?page=${page}`);
 
       if (!response.ok) {
         throw new Error('Network response error');
       }
 
       const data: ShowData[] = await response.json();
 
-      this.setState({
-        shows: data,
-        currentPage: currentPage === 8 ? currentPage - 8 : currentPage + 1,
-        isLoading: false,
-      });
+      setShows(data);
+      setIsLoading(false);
+      setNextPage(page + 1);
+      setPrevPage(page > 0 ? page - 1 : null);
+
+      navigate(`/${ROUTER_PATHS.SHOWS}?page=${encodeURIComponent(page)}`);
     } catch (error) {
-      this.setState({
-        shows: [],
-        isLoading: false,
-        error,
-      });
+      setShows([]);
+      setIsLoading(false);
+      setError(error);
     }
-  }
+  };
 
-  enableButton() {
-    const { shows } = this.state;
-
-    if (shows.length > 20) {
-      this.setState({ isShowMoreButtonDisable: true });
+  const enableButton = () => {
+    if (shows.length >= 20) {
+      setIsShowMoreButtonDisable(true);
     } else {
-      this.setState({ isShowMoreButtonDisable: false });
+      setIsShowMoreButtonDisable(false);
     }
-  }
-
-  showMoreShows() {
-    this.setState((prevState) => ({
-      isMoreShows: !prevState.isMoreShows,
-    }));
-  }
+  };
 
-  getSearchChange() {
+  const getSearchChange = () => {
```

Comment:
Wrap this method in useCallback hook and then you won't need to use ```react-hooks/exhaustive-deps``` exception on 105th line and you'll be able to fill dependency array on 106th

Reply:
Done